src/eat.c

```diff
@@ -890,7 +890,7 @@ givit(register struct permonst *ptr)
 {
     int type = corpse_intrinsic(ptr);
 
-    if (type <= 0) {
+    if (type == 0) {
```

Comment:
Also, the <= in mon_givit ought to be changed too. (A negative value should just pass through that function without doing anything, which is fine because monsters can't get energy or strength, but I don't want to run into this bug again if a third thing is added that pets can get.)

Reply:
Changed it.